clang/lib/Sema/SemaDecl.cpp

```diff
@@ -19553,17 +19555,18 @@ EnumConstantDecl *Sema::CheckEnumConstant(EnumDecl *Enum,
 
         // If we're not in C++, diagnose the overflow of enumerator values,
         // which in C99 means that the enumerator value is not representable in
-        // an int (C99 6.7.2.2p2). However, we support GCC's extension that
-        // permits enumerator values that are representable in some larger
-        // integral type.
-        if (!getLangOpts().CPlusPlus && !T.isNull())
+        // an int (C99 6.7.2.2p2). However C23 permits enumerator values that
+        // are representable in some larger integral type and we allow it in
+        // older language modes as an extension.
+        if (!getLangOpts().CPlusPlus && !getLangOpts().C23 && !T.isNull())
           Diag(IdLoc, diag::warn_enum_value_overflow);
```

Comment:
Should warn_enum_value_overflow also be updated to mention C23?

Reply:
Yeah, probably. Done. Thanks for the catch.